integrations/gitlab_v2/client.py

```diff
@@ -0,0 +1,296 @@
+import asyncio
+from typing import Any, AsyncGenerator
+
+import httpx
+from httpx import Timeout
+from loguru import logger
+
+from rate_limiter import GitLabRateLimiter
+from port_ocean.context.ocean import ocean
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+REQUEST_TIMEOUT: int = 60
+CREATE_UPDATE_WEBHOOK_EVENTS: list[str] = [
+    "open",
+    "reopen",
+    "update",
+    "approved",
+    "unapproved",
+    "approval",
+    "unapproval",
+]
+DELETE_WEBHOOK_EVENTS: list[str] = ["close", "merge"]
+WEBHOOK_EVENTS_TO_TRACK: dict[str, bool] = {
+    "issues_events": True,
+    "merge_requests_events": True,
+}
+WEBHOOK_NAME: str = "Port-Ocean-Events-Webhook"
+
+
+class GitlabClient:
+    def __init__(self, gitlab_host: str, gitlab_token: str) -> None:
+        self.projects_url = f"{gitlab_host}/api/v4/projects"
+        self.merge_requests_url = f"{gitlab_host}/api/v4/merge_requests"
+        self.issues_url = f"{gitlab_host}/api/v4/issues"
+        self.groups_url = f"{gitlab_host}/api/v4/groups"
```

Comment:
Based on these urls, I believe you can attempt to implement methods in this client such that it takes kind from the resync function in main.py and use f string to determine the url to look like `f"{gitlab_host}/api/v4/{kind}s"` this will make the integration support any kind as long as the mapping and blueprint is provided